paddle/fluid/framework/pipeline_trainer.cc

```diff
@@ -141,10 +114,8 @@ std::string PipelineTrainer::GetDumpPath(int tid) {
 void PipelineTrainer::InitDumpEnv() {
   queue_ = paddle::framework::MakeChannel<std::string>();
   // Only set dump channel on the last section
-  for (int j = 0; j < pipeline_num_; ++j) {
-    for (size_t k = 0; k < workers_[section_num_ - 1][j].size(); ++k) {
-      workers_[section_num_ - 1][j][k]->SetChannelWriter(queue_.get());
-    }
+  for (int j = 0; j < section_num_; ++j) {
+    workers_[j]->SetChannelWriter(queue_.get());
   }
   // TODO(hutuxian): should make it as a config
   dump_thread_num_ = 1;
```

Comment:
is this necessary in pipeline trainer?

Reply:
Done. Deleted the unnecessary statements.